src/loaders/FileLoader.js

```diff
@@ -55,225 +55,191 @@ class FileLoader extends Loader {
 
 		}
 
-		// Check for data: URI
-		const dataUriRegex = /^data:(.*?)(;base64)?,(.*)$/;
-		const dataUriRegexResult = url.match( dataUriRegex );
-		let request;
+		// Initialise array for duplicate requests
+		loading[ url ] = [];
+
+		loading[ url ].push( {
+			onLoad: onLoad,
+			onProgress: onProgress,
+			onError: onError,
+		} );
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType
+		// if (this.responseType !== undefined) {
+		//   request.responseType = this.responseType;
+		// }
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/overrideMimeType
+		// if (request.overrideMimeType) {
+		//   request.overrideMimeType(
+		//     this.mimeType !== undefined ? this.mimeType : "text/plain",
+		//   );
+		// }
+
+		// TODO-DefinitelyMaybe: Confirm if Safari can handle Data URIs through fetch
+		// create request
+		const req = new Request( url, {
+			headers: new Headers( this.requestHeader ),
+			credentials: this.withCredentials ? 'include' : 'same-origin',
+			// TODO-DefinitelyMaybe: Could add an abort controller?
+			// signal: this.attachSignal ? this.signal : undefined;
+		} );
```

Comment:
Are you planning on solve these TODOs for this PR or are you leaving them for another?

Reply:
well checking out how Safari handles it is a must before accepting this PR. As for making the fetch an [abortable one](https://developers.google.com/web/updates/2017/09/abortable-fetch), I'd be happy either way. If someone has a strong preference on that right now, please say, otherwise future PR it is.

Reply:
I think adding the ability to cancel a fetch should be added as a separate PR with this one focusing on making sure fetch integration works as expected.

Reply:
Agreed 👍